src/components/ListItem/ListItem.js

```diff
@@ -4,7 +4,7 @@ import { faTrash, faCheckCircle } from '@fortawesome/free-solid-svg-icons';
 
 import styles from './styles.module.css';
 
-const ListItem = ({ actions, keypressHandler, itemKey, text, isCompleted }) => {
+const ListItem = ({ actions, keypressHandler, itemKey, text, isCompleted, isEditable }) => {
   const [value, setValue] = useState(text);
   const mounted = useRef(false);
```

Comment:
I think you'll want to keep the state about whether the item is being edited local to this component, so you can add to this component a new state item, something like this next to the existing `useState()`:
```js
const [isBeingEdited, setIsBeingEdited] = useState();
```

Then when we render the buttons we'll want to make that conditional on the item being edited:

```js
{isBeingEdited ? (
  <>
    <FontAwesomeIcon
      icon={faTrash}
      onClick={() => actions.deleteItemByKey(itemKey)}
      role="button"
      aria-label={`Delete item: ${text}`}
      data-testid={`delete-item-${text}`}
      tabIndex={0}
      className={styles.deleteIcon}
    />
    <FontAwesomeIcon
      icon={faCheckCircle}
      onClick={() => actions.toggleCompletionByKey(itemKey)}
      role="button"
      aria-label={`Added item: ${text}`}
      data-testid={`complete-item-${text}`}
      tabIndex={0}
      className={isCompleted ? (styles.completedIcon) : (styles.completeIcon)}
    />
  </>
) : null}
```

Then for the input box we can set the variable when it is focused or blured:
```
<input
  ... // existing fields
  onFocus={() => setIsBeingEdited(true)}
  onBlur={() => setIsBeingEdited(false)}
```

This will have the problem you mentioned the other day of being unable to click / tap the button because as soon as you do the input will blur and the button disappears, this can be fixed by adding a short delay:
```
onBlur={() => setTimeout(() => setIsBeingEdited(false), 20)}
```

This is a bit hacky, but should work ok

Reply:
Ah ok wicked, thanks I'll give this a go! 

The delay is good to know, I was wanting something like that to make sure the edit mode change happened after everything else!